### PR TITLE
【prim】modify split vjp Concat can't get one input "split_0_tmp_1@GRAD" 

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types

others
### PR changes

others
### Description

pcard-66975
修复组合算子动转静后，spilt 反向找不到输入的问题
eg:
import paddle
from paddle import nn
from paddle import jit
from paddle import fluid

paddle.jit.enable_to_static(True)
#fluid.core.set_prim_eager_enabled(True)
fluid.core._set_prim_all_enabled(True)

x = paddle.randn([4, 1])
y = paddle.randn([4, 1])

x.stop_gradient = False
y.stop_gradient = False

model = nn.Sequential(
    nn.Linear(1, 1),
    nn.Tanh()
)

@jit.to_static
def sol(x, y):
    """abc"""
    z = paddle.concat([x, y], 0)
    out = model(z)
    out0, out1 = paddle.split(out, 2, axis=0)
    g0 = paddle.grad(out0, x)[0]
    g1 = paddle.grad(out1, y)[0]
    return g0, g1

g0, g1 = sol(x, y)
loss = g0.sum() + g1.sum()

loss.backward()
image

### CODE REVIEW

File: python/paddle/fluid/backward.py [hunk 1]
```diff
@@ -1349,7 +1352,28 @@ def update_distop_context(
...
+                if grad_var_name not in grad_name_set:
+                    op_desc = _create_op_desc_(
+                        "fill_any_like",
+                        {"X": [var_name]},
+                        {"Out": [grad_var_name]},
+                        {'value': 1, 'dtype': target_vars[0].dtype},
+                    )
+                    block.desc.append_op().copy_from(op_desc)
+                    clone_block.desc.append_op().copy_from(op_desc)
+                    block.program._sync_with_cpp()
```
Comment (cxxly) [inline python/paddle/fluid/backward.py]: _sync_with_cpp时间复杂度O(n)，开销比较大，移到for循环之外，注意_sync_with_cpp只能sync一段连续子图

File: python/paddle/fluid/backward.py [hunk 2]
```diff
@@ -1349,7 +1352,28 @@ def update_distop_context(
     assert isinstance(rename_var_map, dict)
 
     if core._is_bwd_prim_enabled():
-        composite_block = program.clone().current_block()
+        clone_block = program.clone().current_block()
```
Comment (cxxly) [inline python/paddle/fluid/backward.py]: 整段逻辑是否应该放到append_op之后？

Comment (xiaoguoguo626807) [reply python/paddle/fluid/backward.py]: done

File: python/paddle/fluid/backward.py (hunk 1 above)
Comment (xiaoguoguo626807) [reply python/paddle/fluid/backward.py]: done

**Review by chenwhql — APPROVED** [review]
